ibm/service/contextbasedrestrictions/resource_ibm_cbr_rule.go

```diff
@@ -486,29 +437,102 @@ func resourceIBMCbrRuleUpdate(context context.Context, d *schema.ResourceData, m
 func resourceIBMCbrRuleDelete(context context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	contextBasedRestrictionsClient, err := meta.(conns.ClientSession).ContextBasedRestrictionsV1()
 	if err != nil {
-		return diag.FromErr(err)
+		tfErr := flex.DiscriminatedTerraformErrorf(err, err.Error(), "ibm_cbr_rule", "delete", "initialize-client")
+		log.Printf("[DEBUG]\n%s", tfErr.GetDebugMessage())
+		return tfErr.GetDiag()
 	}
 
 	deleteRuleOptions := &contextbasedrestrictionsv1.DeleteRuleOptions{}
 
 	deleteRuleOptions.SetRuleID(d.Id())
 
-	response, err := contextBasedRestrictionsClient.DeleteRuleWithContext(context, deleteRuleOptions)
+	_, err = contextBasedRestrictionsClient.DeleteRuleWithContext(context, deleteRuleOptions)
 	if err != nil {
-		log.Printf("[DEBUG] DeleteRuleWithContext failed %s\n%s", err, response)
-		return diag.FromErr(fmt.Errorf("DeleteRuleWithContext failed %s\n%s", err, response))
+		tfErr := flex.TerraformErrorf(err, fmt.Sprintf("DeleteRuleWithContext failed: %s", err.Error()), "ibm_cbr_rule", "delete")
+		log.Printf("[DEBUG]\n%s", tfErr.GetDebugMessage())
+		return tfErr.GetDiag()
 	}
 
 	d.SetId("")
 
 	return nil
 }
 
-func resourceIBMCbrRuleMapToRuleContext(modelMap map[string]interface{}) (*contextbasedrestrictionsv1.RuleContext, error) {
+func ResourceIBMCbrRuleSetData(rule *contextbasedrestrictionsv1.Rule, response *core.DetailedResponse, d *schema.ResourceData) error {
+	if !core.IsNil(rule.Description) {
+		if err := d.Set("description", rule.Description); err != nil {
+			return fmt.Errorf("Error setting description: %s", err)
+		}
+	}
+	if !core.IsNil(rule.Contexts) {
+		contexts := []map[string]interface{}{}
+		for _, contextsItem := range rule.Contexts {
+			contextsItemMap, err := ResourceIBMCbrRuleRuleContextToMap(&contextsItem) // #nosec G601
+			if err != nil {
+				return fmt.Errorf("Error map rule context: %s", err)
+			}
+			contexts = append(contexts, contextsItemMap)
+		}
+		if err := d.Set("contexts", contexts); err != nil {
```

Comment:
contexts is no longer set to an empty list if not specified?  What is the reason for this change?

Reply:
The `ResourceIBMCbrRuleSetData` is called after create or read operations. The API should return an empty context list if not specified. The changes came from the generated code, and it should be functionally equivalent to the previous code.
Anyway, I changed it back to its previous one.